### PR TITLE
proto(dropdown): @mention scenario

### PR DESCRIPTION
**Pull request by bmdalex**
### Description

This PR attempts to create a prototype to show users how they can mention people using `@` key in a rich text editor.

I am using a `contentEditable` div as rich text input and the `Dropdown` component to leverage the list of users and filtering through them. The rest of the functionality is in the prototype section 

### Screenshot
screen recording 2019-02-14 at 16 51 35

### Issues:
- #900 and issue in description - **blocker**
- use `createPortal` instead of mounting another element https://github.com/stardust-ui/react/pull/901#discussion_r256901115

### CODE REVIEW

File: docs/src/prototypes/dropdowns/AsyncDropdownSearch.tsx [hunk 1]
```diff
@@ -34,25 +34,55 @@ const createEntry = (): Entry => ({
...
   }
 
-  handleSearchQueryChange = (e: React.SyntheticEvent, { searchQuery }: DropdownProps) => {
+  private handleSearchQueryChange = (e: React.SyntheticEvent, { searchQuery }: DropdownProps) => {
     this.setState({ searchQuery })
     this.fetchItems()
   }
 
-  fetchItems = () => {
+  private fetchItems = () => {
```
Comment (layershifter) [inline docs/src/prototypes/dropdowns/AsyncDropdownSearch.tsx:85]: What is purpose of these changes? Do we need them?
We don't have an agreement on visibility modifiers so let's revert these changes.

File: docs/src/prototypes/dropdowns/inputWithDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      />
+    )
+  }
+
+  private showDropdown = () => {
+    this.dropdownExists = true
+    insertNodeAtCursorPosition({ id: this.mountNodeId })
+
+    const node = this.getMountNode()
+    ReactDOM.render(
```
Comment (layershifter) [inline docs/src/prototypes/dropdowns/inputWithDropdown.tsx:49]: Can you please why we can't use `createPortal()`/`PortalInner` component there? What is a reason for another React's root?

File: docs/src/prototypes/dropdowns/utils.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    return
+  }
+
+  const elem = document.createElement('span')
+  elem.id = id
+  if (text) {
+    elem.innerText = text
+  }
+  range.insertNode(elem)
+}
```
Comment (layershifter) [inline docs/src/prototypes/dropdowns/utils.ts:36]: Is there any third party dependency that does this work? That is fully tested and supported?
Why we can't use [`caret-pos`](https://www.npmjs.com/package/caret-pos)?

File: docs/src/prototypes/dropdowns/AsyncDropdownSearch.tsx (hunk 1 above)
Comment (Bugaa92) [reply docs/src/prototypes/dropdowns/AsyncDropdownSearch.tsx]: > We don't have an agreement on visibility modifiers so let's revert these changes.

I don't get this part, it's pretty obvious to me that in Typescript class members that are not meant to be used outside the scope of the class are private.

> What is purpose of these changes? Do we need them?

I will revert the changes however because the prototype section does not have the JS version so you might argue the change does not bring any benefit here.

File: docs/src/prototypes/dropdowns/inputWithDropdown.tsx (hunk 2 above)
Comment (bmdalex) [reply docs/src/prototypes/dropdowns/inputWithDropdown.tsx:49]: opened this PR agains wrong branch, moved to: https://github.com/stardust-ui/react/pull/931/files#r258166628

File: docs/src/prototypes/dropdowns/utils.ts (hunk 3 above)
Comment (bmdalex) [reply docs/src/prototypes/dropdowns/utils.ts:36]: opened this PR agains wrong branch, moved to: https://github.com/stardust-ui/react/pull/931#discussion_r258165673

File: docs/src/prototypes/dropdowns/AsyncDropdownSearch.tsx (hunk 1 above)
Comment (bmdalex) [reply docs/src/prototypes/dropdowns/AsyncDropdownSearch.tsx:85]: opened this PR agains wrong branch, moved to: https://github.com/stardust-ui/react/pull/931/files#r258164858

**Comment by bmdalex** [comment]
opened this PR agains wrong branch, moved to #931
